### DISCUSSION TITLE
Running on Unix domain socket

### DISCUSSION THREAD

**Discussion by gioboske** [zauberzeug/nicegui#3898] [Ideas / Feature Requests]
When running in native mode, it could be useful to run `NiceGUI` server over Unix domain socket, instead of `localhost`. For example, to support access control using file permissions in a multi-users system.

`uvicorn` already allows to bind to a Unix domain socket (UDS), so I guess it is a matter of make `pywebview` communicating with a server running over UDS. Something similar to send requests with `curl` by using `--unix-socket <path>` option.

Some time ago I've opened a similar request on `pywebview` repository (r0x0r/pywebview#1393), but it is not planned to implement this feature, at least in the near feature.

As far as I know, there is no standard url scheme for UDS. In fact, similar feature request for Chromium lead to a WONTFIX also because of the lack of a standard url scheme (just as reference, here the [feature request for Chromium](https://issues.chromium.org/issues/40402523) and a similar discussion on [Bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1688774)).

Maybe, someone (more skilled than me) could propose a clever way to proxy all `NiceGUI`/`pywebview` connections to the Unix socket.

**Comment by rodja** [comment]
You are right, it would be great to have this this feature. As far as I know, `pywebview` uses a standard browser like Chromium to render the content.
If you could somehow proxy the connection to a port, the benefits of not having a port would be lost...
